### DISCUSSION TITLE
Considering dropping support for Python 3.7

### DISCUSSION THREAD

**Discussion by anjakefala** [saulpw/visidata#2004] [General]
I keep running into features that are not supported in Python 3.7.

The latest example is the reason the tests are currently failing. `start` only got added in `sum()` in Python 3.8: https://docs.python.org/3/library/functions.html#sum. 

The [NumPy support table](https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule) is one I see other projects employ. It says that NumPy dropped support for Python 3.7 on Dec 26, 2021, which was quite a long time ago.

What do we think about dropping that support? Is anyone still on Python 3.7?

**Comment by daviewales** [comment]
Hmm... I think I was partly responsible for the `start` argument being used!
I had a look at the Debian Python package, and you have to go back 2 versions before the current stable version of Debian to get to Python 3.7:
https://packages.debian.org/buster/python3

Current stable has Python 3.11:
https://packages.debian.org/bookworm/python3

Note that Buster is still LTS supported until June 2024:
https://wiki.debian.org/LTS

On the other hand, regular non-LTS support for Buster ended in September 2022:
https://wiki.debian.org/DebianReleases#Production_Releases

Note also that Python 3.7 is no longer supported as of June 2023:
https://devguide.python.org/versions/
https://peps.python.org/pep-0537/

So I'd say it's _reasonably_ safe to drop support.

**Reply by daviewales** [reply]
Oh, I see this has already been done!

**Reply by anjakefala** [reply]
But, I appreciate the support that it was a reasonable decision!

**Reply by cool-RR** [reply]
I second that decision, and I generally support aggressively dropping support for old Python versions in the future (e.g. when you start feeling pressure to drop 3.8, 3.9 etc). People who want to use old Python versions can continue using old VisiData versions.
